docs/reference/ingest/processors/kv.asciidoc

```diff
@@ -17,8 +16,11 @@ For example, if you have a log message which contains `ip=1.2.3.4 error=REFUSED`
 --------------------------------------------------
 // NOTCONSOLE
 
+TIP: Using the KV Processor can result in field names that you cannot control. Consider using the <<flattened>> datatype instead, which maps an entire object as a single field.
+While a flattened object provides only a single field to search on, the object's contents can still be searched using simple queries and aggregations.
```

Comment:
To me the phrase "While a flattened object provides only a single field to search on" could be confusing -- it could suggest that you can only search the root field. What would you think of this tweak: "Using the KV Processor can create a large number of field names that you don't control. Consider using the flattened datatype instead, which maps an entire object as a single field and allows for simple searches over its contents."

Reply:
Agreed! I like your edit, which focuses on the positive (what the user _can_ do) of the `flattened` object, rather than its limitations.